.github/workflows/deploy-site.yml

```diff
@@ -0,0 +1,32 @@
+name: Build and deploy SSG website
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Install AsciiDoctor
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y asciidoctor
+
+      - name: Build AsciiDoc
+        run: |
+          asciidoctor supplementary_style_guide/main.adoc -o index.html
+          mkdir docs
+          cp -r supplementary_style_guide/images/ docs/images
+          mv index.html docs/
+
+      - name: Publish to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
@aireilly is this GITHUB_TOKEN something we have to configure somewhere?

Reply:
Should be available by default.

Reply:
FYI for anyone else curious, it's this: https://docs.github.com/en/actions/security-guides/automatic-token-authentication